src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/extension/env/EnvExtension.groovy

```diff
@@ -93,6 +106,22 @@ class EnvExtension extends AbstractGlobalExtension implements SpringContextListe
             throw new RuntimeException("Provided profile '$profile' is unknown. Select one of the following profiles:" +
                     " hardware, virtual")
         }
+        if((profile == "hardware" && enable) || (profile == "virtual" && !isTopologyRebuildRequired && enable)) {
```

Comment:
It's useful to move complicated conditions into boolean methods with readable name. E.g. this one can be moved into something like `boolean isHealthcheckNeeded(String profile, Boolean enable, Boolean isTopologyRebuildRequired)`.